experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -39,6 +39,12 @@ export function setTreeNode(target: any, treeNode: TreeNode) {
 	});
 }
 
+/**
+ * Caches the proxy object that is created for each node.
+ * This saves memory, improves performance, and guarantees that all proxies retrieved for a given {@link TreeNode} are referentially equal.
+ */
+const nodeProxyCache = new WeakMap<TreeNode, ProxyNode<TreeSchema>>();
```

Comment:
as discussed offline, we should go with symbols over WeakMap for read perf

Reply:
Done.